exercises/regularization/ex_rnw/ex_lasso_regularization.Rnw

```diff
@@ -0,0 +1,128 @@
+
+Consider the regression learning setting, i.e., $\mathcal{Y} = \R,$ and feature space $\Xspace = \R^p.$
+Let the hypothesis space be the linear models:
+
+$$
+  \Hspace = \{ \fx = \thetab^\top \xv ~|~  \thetab \in \R^p  \}. 
+$$
+Suppose your loss function of interest is the L2 loss $\Lxy= \frac12 \big(y-\fx\big)^2.$ Consider the $L_1$-regularized empirical risk of a model $\fxt$ (i.e., Lasso regression):
+$$
+	\riskrt = \risket + \lambda \|\thetab\|_1 =  \frac12 \sum_{i=1}^n \left( \yi - \thetab^\top \xi \right)^2 + \lambda \sum_{i=1}^p  |\theta_i|.
+$$
+Assume that ${\Xmat}^T \Xmat = \id,$ which holds if $\Xmat$ has orthonormal columns. Show that the minimizer $\thetah_{\text{Lasso}} = (\thetah_{\text{Lasso},1},\ldots,\thetah_{\text{Lasso},p})^\top$ is given by
+
+$$	
+	\thetah_{\text{Lasso},i} = sgn(\thetah_{i}) \max\{|\thetah_{i}| - \lambda,0 \}, \quad i=1,\ldots,p,	
+$$
+
+where $\thetabh = (\thetah_{1},\ldots,\thetah_{p})^\top = ({\Xmat}^T \Xmat)^{-1} \Xmat^T\yv$ is the minimizer of the unregularized empirical risk (w.r.t.\ the L2 loss). For this purpose, use the following steps:
+\begin{itemize}	
+	\item
+		Derive that 
+	
+		$$	
+			\argmin_{\thetab} \riskrt  = 	\argmin_{\thetab} \sum_{i=1}^p	- \thetah_{i}\theta_i + \frac{\theta_i^2}{2} + \lambda|\theta_i|.
+		$$
+	
+		\lz
+		\lz
+		\lz
+		\lz
+		\lz
+		\lz
+		\lz
+		\lz
+		\lz
+	
+	
+	
+	\item 
+	
+		Note that the minimization problem on the right-hand side of (i) can be written as $\sum_{i=1}^p g_i(\theta_i),$ where 
+		$$ 
+			g_i(\theta) =  - \thetah_{i}\theta + \frac{\theta^2}{2} + \lambda |\theta|. 
```

Comment:
Perhaps g_i(\theta_i)? Is not wrong as-is, but perhaps clearer?

Reply:
Changed